samples/spring-boot-security/src/main/java/sample/Application.java

```diff
@@ -2,57 +2,45 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
 
 @SpringBootApplication
 public class Application {
 
     @Configuration
-    protected static class ApplicationSecurity extends WebSecurityConfigurerAdapter {
+    protected static class ApplicationSecurity {
 
-        @Override
-        protected void configure(HttpSecurity http) throws Exception {
+        @Bean
+        protected SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
             // @formatter:off
             http
                     .authorizeRequests()
-                        .anyRequest()
-                        .authenticated()
-                        .and()
+                    .anyRequest()
+                    .authenticated()
+                    .and()
                     .csrf()
-                        .disable()
+                    .disable()
                     .formLogin()
-                        .and()
+                    .and()
                     .logout();
             // @@formatter:on
+            return http.build();
         }
 
-        @Override
-        public void configure(AuthenticationManagerBuilder auth) throws Exception {
-            // @formatter:off
-            auth.inMemoryAuthentication()
-                    .withUser("admin")
-                        .password("pwd")
-                        .roles("ADMIN", "USER")
-                        .and()
-                    .withUser("user1")
-                        .password("pwd")
-                        .roles("USER")
-                        .and()
-                    .withUser("user2")
-                        .password("pwd")
-                        .roles("USER")
-                        .and()
-                    .withUser("user3")
-                        .password("pwd")
-                        .roles("USER")
-                        .and()
-                    .withUser("user4")
-                        .password("pwd")
-                        .roles("USER");
-            // @formatter:on
+        @Bean
+        public InMemoryUserDetailsManager userDetailsService() {
+            UserDetails admin = User.withDefaultPasswordEncoder().username("admin").password("pwd").roles("ADMIN", "USER").build();
```

Comment:
The method `withDefaultPasswordEncoder` is marked as deprecated where the explanation says that it recommended just for demo purposes but not for production. We use it exactly for demo purposes so my suggestion is to add annotation `@SupressWarnings("deprecation")` on top of our method `userDetailsService`